apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,521 @@
+	if (ret < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Timer : %d\n", errno);
+		ret = -1;
+		close(frt_fd);
+	}
+
+	close(frt_fd);
+	return ret;
+
+error_with_frt_status:
+	fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+	ret = -1;
+	goto close_with_frt;
+close_with_fd:
+	close(fd);
+close_with_frt:
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Free Run Timer: %d\n", errno);
+	}
+
+	close(frt_fd);
+
+	return ret;
+}
+
+static void show_usage() {
+	printf("\nusage: fs_performance [Timer device no.] [Test case] [File size(bytes)] [Buffer size(bytes)]\n");
+	printf("Timer Device no. :         FRT Timer device\n");
+	printf("Test Case        :         Test case to be executed\n");
+	printf("      1. File Creation Test : Creates a file of size \"File size\" in one go.\n");
+	printf("      2. Overwrite Test : Overwrite a file from start with buffer size \"Buffer size\".\n");
+	printf("      3. Append Test : Append 64 kb data to a file of size \"File size\".\n");
+	printf("      4. Read Test : Read a file of size \"File size\" with buffer size \"Buffer size\".\n");
+	printf("      5. Garbage Collection Effect Test : Create a file and unlink it to trigger GC.\n");
+	printf("File Size        :         File size(bytes) less than equal to 64kb\n");
+	printf("Buffer Size      :         Buffer size(bytes) less than equal to File size\n");
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int fs_performance_main(int argc, char *argv[])
+#endif
+{
+	int option = -1;
+	int ret = 0;
+	int timer_devno = 0;
+	int selected_opt;
+	g_fsz = MAX_FILE_SIZE;
+	g_bsz = MAX_RD_BUF_SIZE;
+
+	if (argc == 2 && !strncmp(argv[1], "--help", strlen("--help") + 1)) {
+		show_usage();
+		return OK;
+	}
+
+	if (argc > 4) {
+		timer_devno = atoi(argv[1]);
+		option = atoi(argv[2]);
+		if (option < 1 || option > 5) {
+			printf("Test case option should be within 1-5\n");
+			goto errout;
+		}
+
+		g_fsz = atoi(argv[3]);
+		if (g_fsz > MAX_FILE_SIZE) {
+			printf("File size should be less than or equal to %d bytes\n", MAX_FILE_SIZE);
+			goto errout;
+		}
+
+		if (argc == 5) {
+			g_bsz = atoi(argv[4]);
+			if (g_bsz > MAX_BUF_SIZE || g_bsz > g_fsz) {
+				printf("Buffer size should be less than or equal to Buffer size: %d bytes & File size: %d bytes\n", MAX_BUF_SIZE, g_fsz);
+				goto errout;
+			}
+		}
+	} else {
+		ret = -1;
+		goto errout;
+	}
+
+	snprintf(g_timer_path, _POSIX_PATH_MAX, TIMER_DEVNAME, timer_devno);
+	snprintf(g_file[0], TEST_FILE_NAME_LEN_MAX, "%s%d", TEST_FILE, 0);
+
+	switch (option) {
+
+		case 1: selected_opt = TEST_FLAG_FILE_CREATION;
+			break;
+
+		case 2: selected_opt = TEST_FLAG_OVERWRITE;
+			break;
+
+		case 3: selected_opt = TEST_FLAG_APPEND;
+			break;
+
+		case 4: selected_opt = TEST_FLAG_READ;
+			break;
+
+		case 5: selected_opt = TEST_FLAG_GC_EFFECT;
+			break;
+
+		default:
+			goto errout;
+
+	}
+
+	ret = do_test(selected_opt);
```

Comment:
You use the global variables for 3rd and 4th arg. But you do the local variable for 2nd arg. Could you let me know why you use them with different ways?